backend/plugin/db/mysql/query.go

```diff
@@ -55,3 +58,48 @@ func getStatementWithResultLimitForTiDB(singleStatement string, limitCount int)
 	}
 	return "", nil
 }
+
+// singleStatement must be a selectStatement for mysql.
+func getStatementWithResultLimitForMySQL(singleStatement string, limitCount int) (string, error) {
+	list, err := mysqlparser.ParseMySQL(singleStatement)
+	if err != nil {
+		return "", err
+	}
+
+	listener := &mysqlRewriter{
+		limitCount:     limitCount,
+		outerMostQuery: true,
+	}
+
+	for _, stmt := range list {
+		listener.rewriter = *antlr.NewTokenStreamRewriter(stmt.Tokens)
+		antlr.ParseTreeWalkerDefault.Walk(listener, stmt.Tree)
+		if listener.err != nil {
+			return "", errors.Wrapf(listener.err, "statement: %s", singleStatement)
+		}
+	}
+	return listener.rewriter.GetTextDefault(), nil
+}
+
+type mysqlRewriter struct {
+	*mysql.BaseMySQLParserListener
+
+	rewriter       antlr.TokenStreamRewriter
+	err            error
+	outerMostQuery bool
+	limitCount     int
+}
+
+func (r *mysqlRewriter) EnterQueryExpression(ctx *mysql.QueryExpressionContext) {
+	if !r.outerMostQuery {
+		return
+	}
+	r.outerMostQuery = false
+	limitClause := ctx.LimitClause()
+	if limitClause != nil {
+		// limit clause already exists.
+		return
+	}
+
+	r.rewriter.InsertAfterDefault(ctx.GetStop().GetTokenIndex(), fmt.Sprintf(" LIMIT %d", r.limitCount))
```

Comment:
We define `QueryExpression` as:
```
queryExpression:
    (withClause)? (
        queryExpressionBody orderClause? limitClause?
        | queryExpressionParens orderClause? limitClause?
    ) (procedureAnalyseClause)?
;
```
So, we cannot add `LIMIT` clause after `ctx.GetStop().GetTokenIndex()`, it may have `procedureAnalyseClause`

Reply:
maybe we can filter by `queryExpressionBody` and `queryExpressionParens` ?

Reply:
We don't need it. We can simply check that it has `orderClause` or not. If it has, insert it after `orderClause`. If not, insert it after `queryExpressionBody` or `queryExpressionParens`.